setup.py

```diff
@@ -80,6 +82,29 @@ def get_version():
     return locals()['__version__']
 
 
+def make_cuda_ext(name, module, sources):
+
+    return CUDAExtension(
+        name='{}.{}'.format(module, name),
+        sources=[os.path.join(*module.split('.'), p) for p in sources])
+
+
+def make_cython_ext(name, module, sources):
+    from Cython.Build import cythonize
+    import numpy as np
```

Comment:
This method will be called before installing dependencies, so that an error will still be raised.
A feasible solution may be installing cython before setting up mmdetection.

Reply:
You're right, it doesn't make a difference putting it there.

The `INSTALL.md` already mentions installing cython. Since we don't list pytorch in the `install_requires` field, we can't simply pip install the package anyway without prior installation of pytorch and cython. So how do you think we should handle those two dependencies?

Reply:
Yes it seems nontrivial to avoid this dependency, so we may just assume that users have followed the installation guide to install cython first. Thus I think it's ok to move these import statements outside the method. Thanks!

Reply:
Sounds good! Done in 18c0e5e